lib/src/main/java/io/ably/lib/http/Http.java

```diff
@@ -81,19 +116,25 @@
 	 *     Public API
 	 *************************/
 
-	public Http(AblyRest ably, ClientOptions options) {
-		this.ably = ably;
+	public Http(ClientOptions options, Auth auth) {
+		this.options = options;
+		this.auth = auth;
+		this.host = options.restHost;
 		this.scheme = options.tls ? "https://" : "http://";
 		this.port = Defaults.getPort(options);
 	}
 
-	private String getPrefHost() {
-		if(ably instanceof AblyRealtime) {
-			Connection connection = ((AblyRealtime)ably).connection;
-			if(connection.state == ConnectionState.connected)
-				return connection.connectionManager.getHost();
-		}
-		return Defaults.getHost(ably.options);
+	public boolean isFallbackSupported() {
+		return host.matches(PATTERN_HOST_PROD);
+	}
+
+	/**
+	 * Sets host for this HTTP client
+	 *
+	 * @param host URL string
+     */
```

Comment:
This is displayed similar within IDE. This is why I am pushing `Reformating code` change. Currently I am unable to detect these mistakes with bare eyes, and these are only visible when github is using different spacing for tabs